### PR TITLE
Optical Circuit Recipes

### PR DESCRIPTION
**Pull request by GDCloudstrike**
adds recipes for more optical circuits parts and the circuits themselves, not finished yet, more stuff to come

### CODE REVIEW

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 1]
```diff
@@ -26479,4 +26481,166 @@ private static final void registerNEIPlanetRecipes() {
...
+
+    private static final void registerOpticalComponentRecipes() {
+        // Optical Boule
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[] {
+                    ItemList.Circuit_Silicon_Ingot5.get(1L), // Americium Boule
+                    GT_Utility.copyAmount(
+                            0L,
+                            GT_ModHandler.getModItem(
+                                    "bartworks", "gt.bwMetaGeneratedlens", 1L, 36)), // Magneto resonatic lens
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26493]: Can do Loader.isModLoaded("bartworks") and then use WerkstoffLoader.Material.get(OrePrefixes.lens, 0) Laser engravers don't consume lenses unless intended

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 2]
```diff
@@ -26479,4 +26481,166 @@ private static final void registerNEIPlanetRecipes() {
...
+                    GT_Utility.copyAmount(
+                            0L,
+                            GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 1L, 23)), // Fayalite lens
+                    GT_Utility.copyAmount(0L, CustomItemList.MysteriousCrystalLens.get(1))
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26497]: use the above mentioned method for these two.

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 3]
```diff
@@ -26479,4 +26481,166 @@ private static final void registerNEIPlanetRecipes() {
...
+                : FluidRegistry.getFluid("radon");
+
+        // Photonically Enhanced Wafer
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[] {
+                    ItemList.Circuit_Silicon_Wafer6.get(1L), // Photonically Prepared Wafer
+                    Materials.Glowstone.getNanite(1),
+                    GT_Utility.copyAmount(
+                            0L,
+                            GT_ModHandler.getModItem("supersolarpanel", "solarsplitter", 1L, 0)) // Solar Light Splitter
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26540]: same thing, but here you use GT_ModHandler.getModItem("supersolarpanel", "solarsplitter", 0L, 0)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 4]
```diff
@@ -26479,4 +26481,166 @@ private static final void registerNEIPlanetRecipes() {
...
+                ItemList.Optically_Perfected_CPU.get(1L),
+                20 * 20,
+                1_966_080,
+                true);
+
+        // Optical CPU Containment Housing
+        GT_Values.RA.addFormingPressRecipe(
+                new ItemStack[] {
+                    ItemList.Circuit_Board_Optical.get(1L),
+                    GT_ModHandler.getModItem("miscutils", "itemFoilBotmium", 1L),
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26570]: can do Loader.isModLoader("miscutils") and then use ALLOYS (from GT++, it should be in as a compile only)
GT_OreDictUnificator(OrePrefixes.foil, ALLOYS.Botmium, 1)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 5]
```diff
@@ -26484,4 +26486,209 @@ private static final void registerNEIPlanetRecipes() {
...
+            GT_Values.RA.addDistillationTowerRecipe(
+                    Materials.Grade6PurifiedWater.getFluid(1000L),
+                    new FluidStack[] {
+                        Materials.Grade7PurifiedWater.getFluid(1000L),
+                    },
+                    null,
+                    500 * 10,
+                    1_966_080);
+            GT_Values.RA.addChemicalBathRecipe(
+                    GT_Utility.copyAmount(0L, Materials.Silver.getNanite(1)),
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26644]: Can just do Materials.Silver.getNanite(0)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 6]
```diff
@@ -26484,4 +26486,209 @@ private static final void registerNEIPlanetRecipes() {
...
+
+            // Photonically Enhanced Wafer
+            GT_Values.RA.addLaserEngraverRecipe(
+                    new ItemStack[] {
+                        ItemList.Circuit_Silicon_Wafer6.get(1L), // Photonically Prepared Wafer
+                        Materials.Glowstone.getNanite(1),
+                        GT_Utility.copyAmount(
+                                0L,
+                                GT_ModHandler.getModItem(
+                                        "supersolarpanel", "solarsplitter", 1L, 0)) // Solar Light Splitter
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26543]: Does getting an item with 0 item count not work? it should

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 7]
```diff
@@ -26484,4 +26486,209 @@ private static final void registerNEIPlanetRecipes() {
...
+    private static final void registerOpticalComponentRecipes() {
+
+        if (Loader.isModLoaded("bartworks")) {
+            // Optical Boule
+            GT_Values.RA.addLaserEngraverRecipe(
+                    new ItemStack[] {
+                        ItemList.Circuit_Silicon_Ingot5.get(1L), // Americium Boule
+                        WerkstoffLoader.MagnetoResonaticDust.get(OrePrefixes.lens, 0), // Magneto resonatic lens
+                        WerkstoffLoader.Fayalit.get(OrePrefixes.lens, 0), // Fayalite lens
+                        GT_Utility.copyAmount(0L, CustomItemList.MysteriousCrystalLens.get(1))
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26499]: can just get an item with a stack size of 0

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java (hunk 5 above)
Comment (GDCloudstrike) [reply src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26644]: doesn't work sadly

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 8]
```diff
@@ -26616,7 +26616,7 @@ private static final void registerOpticalComponentRecipes() {
                     245_760,
                     true);
             GT_Values.RA.addMultiblockChemicalRecipe(
-                    new ItemStack[] {Materials.Zeolite.getDust(0), GT_Utility.getIntegratedCircuit(1)},
+                    new ItemStack[] {GT_Utility.copyAmount(0, Materials.Zeolite.getDust(5), GT_Utility.getIntegratedCircuit(1))},
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:26619]: okay it appears i found the problem.
GT_OreDictUnificator.get(OrePrefixes.dust, Materials.Zeolite, 0, false) will fix it

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**